src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/jvm/sizing/JvmScaleUpPolicy.java

```diff
@@ -0,0 +1,145 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Action;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.SizeUpJvmAction;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.deciders.DecisionPolicy;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.configs.JvmScaleUpPolicyConfig;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.aggregators.SlidingWindow;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.aggregators.SlidingWindowData;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.ResourceFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotClusterSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotNodeSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.RcaConf;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.jvmsizing.LargeHeapClusterRca;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class JvmScaleUpPolicy implements DecisionPolicy {
+
+  private final LargeHeapClusterRca largeHeapClusterRca;
+  private AppContext appContext;
+  private RcaConf rcaConf;
+  private PerNodeSlidingWindow perNodeSlidingWindow;
+  private long evalFrequency;
+  private long counter;
+  private int unhealthyNodePercentage;
+  private int minimumMinutesUnhealthy;
+
+  public JvmScaleUpPolicy(final LargeHeapClusterRca largeHeapClusterRca,
+      final long policyEvaluationFrequency) {
+    this.largeHeapClusterRca = largeHeapClusterRca;
+    this.evalFrequency = policyEvaluationFrequency;
+    this.counter = 0;
+    this.perNodeSlidingWindow = new PerNodeSlidingWindow(4, TimeUnit.DAYS);
+  }
+
+  @Override
+  public List<Action> evaluate() {
+    counter++;
+    addToSlidingWindow();
+    if (counter == evalFrequency) {
+      counter = 0;
+      return evaluateAndEmit();
+    }
+
+    return Collections.emptyList();
+  }
+
+  private void addToSlidingWindow() {
+    long currTime = System.currentTimeMillis();
+    if (largeHeapClusterRca.getFlowUnits().isEmpty()) {
+      return;
+    }
+    ResourceFlowUnit<HotClusterSummary> flowUnit = largeHeapClusterRca.getFlowUnits().get(0);
+
+    if (flowUnit.getSummary() == null) {
+      return;
+    }
+    List<HotNodeSummary> hotNodeSummaries = flowUnit.getSummary().getHotNodeSummaryList();
+    hotNodeSummaries.forEach(hotNodeSummary -> {
+      NodeKey nodeKey = new NodeKey(hotNodeSummary.getNodeID(), hotNodeSummary.getHostAddress());
```

Comment:
I get what you're saying, but we defensively create a new ArrayList for node summaries. I've added a @NonNull annotation on the getter to enforce static checking of (non)nullness so that others can also safely use the list without adding null checks.